src/ansys/motorcad/core/rpc_client_core.py

```diff
@@ -185,22 +192,30 @@ def __init__(
         if self.reuse_parallel_instances is True:
             self._open_new_instance = False
 
-        if _HAS_PIM and pypim.is_configured():
+        if self._url != "":
+            # Don't want to start a Motor-CAD instance
+            # Already have full url in _get_url()
+            pass
+        elif _HAS_PIM and pypim.is_configured():
             # Start with PyPIM if the environment is configured for it
             self._launch_motorcad_remote()
         else:
             # run/connect to motor-cad on local machine
             self._launch_motorcad_local(port)
 
         # Check for response
-        if self._wait_for_response(2) is True:
+        if self._wait_for_response(self._timeout) is True:
             self._connected = True
 
             # Store Motor-CAD version number for any required backwards compatibility
             self.program_version = self._get_program_version()
 
             self.pid = self.get_process_id()
 
+            if (SERVER_IP == LOCALHOST_ADDRESS) and TRY_RESOLVE_LOCALHOST:
+                # Try to resolve localhost to speed up function calls
+                self._resolve_localhost()
```

Comment:
Not covered directly as if this fails then all other tests will fail, so it will be obvious this is a problem

Reply:
As discussed, think it would be useful to test this independently. If the resolving fails then other tests may still pass but might be slower as they will just use the localhost address.